src/unescape.c

```diff
@@ -236,13 +237,14 @@ unescape(yp_parser_t *parser, uint8_t *dest, size_t *dest_length, const uint8_t
 
                 unicode_cursor += yp_strspn_whitespace(unicode_cursor, end - unicode_cursor);
 
-                while ((*unicode_cursor != '}') && (unicode_cursor < end)) {
+                while ((unicode_cursor < end) && (*unicode_cursor != '}')) {
```

Comment:
Why this change?

Reply:
We need to check if `unicode_cursor` is past `end` _before_ dereferencing it, otherwise we could be illegally accessing memory past the end of the buffer.

Reply:
Hmm, looking at it more closely and this whole loop construct is more complicated than we need. Let me take another pass at cleaning it up.